server/static/js/toggle-ownership.component.js

```diff
@@ -15,7 +15,27 @@ AFRAME.registerComponent('toggle-ownership', {
 
     if (NAF.utils.isMine(this.el)) {
       this.updateColor();
+    } else {
+      this.el.setAttribute('material', 'opacity', 0.5);
     }
+
+    // Opacity is not a networked attribute, but change it based on ownership events
+    this.networkedEl = NAF.utils.getNetworkedEntity(this.el);
+    
+    this.networkedEl.addEventListener("ownership-gained", e => {
+      this.el.setAttribute('material', 'opacity', 1);
+    });
+
+    this.networkedEl.addEventListener("ownership-lost", e => {
+      this.el.setAttribute('material', 'opacity', 0.5);
+    });
+
+    this.networkedEl.addEventListener("ownership-changed", e => {
+      this.el.setAttribute('material', 'opacity', 0.8);
+      setTimeout(() => {
+        this.el.setAttribute('material', 'opacity', 0.5);
```

Comment:
Race condition here since this could end up firing after one of the other events. You would need to store the timeout and clear it in the other events like:

```
let blinkTimeout;
this.networkedEl.addEventListener("ownership-gained", e => {
  clearTimeout(blinkTimeout);
  this.el.setAttribute("material", "opacity", 1);
});

this.networkedEl.addEventListener("ownership-lost", e => {
  clearTimeout(blinkTimeout);
  this.el.setAttribute("material", "opacity", 0.5);
});

this.networkedEl.addEventListener("ownership-changed", e => {
  this.el.setAttribute("material", "opacity", 0.8);
  blinkTimeout = setTimeout(() => {
    this.el.setAttribute("material", "opacity", 0.5);
  }, 200);
});
```

Or if we change the events like I suggested above:
```
let blinkTimeout;
this.networkedEl.addEventListener("ownership-changed", e => {
  clearTimeout(blinkTimeout);
  if(e.newOwner === NAF.clientId) {
    this.el.setAttribute("material", "opacity", 1);
  } else if(e.oldOwner === NAF.clientId) {
    this.el.setAttribute("material", "opacity", 0.5);
  } else {
    this.el.setAttribute("material", "opacity", 0.8);
    blinkTimeout = setTimeout(() => {
      this.el.setAttribute("material", "opacity", 0.5);
    }, 200);
  }
})